### ISSUE TITLE
Pytorch 2.0 compiled model complains stride mismatch about pixel_shuffle

### ISSUE THREAD

**Issue by silent-monkey** [pytorch/pytorch#91551]
### 🐛 Describe the bug

Pytorch 2.0 model compiled by `torch.compile` complaining mismatch stride when evaling (on CPU).

Here's a minimized case extracted from my model:

```python
import torch
from torch import nn
from torch.nn import functional as F

g = nn.Sequential(
    nn.Conv2d(16, 64, 3, 1, 1),
    nn.PixelShuffle(2),
    nn.ReLU(),
}
g.eval()

gg = torch.compile(g, fullgraph=True)

with torch.no_grad():
    # `no_grad` is essential to trigger the bug
    print(gg(torch.rand((1, 16, 64, 72))))
```

Full backtrace:
```
Traceback (most recent call last):
  File "/home/fsp/work/myEDVR/bug.py", line 65, in <module>
    print(gg(torch.rand((1, 16, 64, 72))))
  File "/home/fsp/.virtualenvs/nn/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1482, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/fsp/.virtualenvs/nn/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 83, in forward
    return self.dynamo_ctx(self._orig_mod.forward)(*args, **kwargs)
  File "/home/fsp/.virtualenvs/nn/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 212, in _fn
    return fn(*args, **kwargs)
  File "/home/fsp/.virtualenvs/nn/lib/python3.10/site-packages/torch/nn/modules/container.py", line 215, in forward
    def forward(self, input):
  File "/home/fsp/.virtualenvs/nn/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py", line 212, in _fn
    return fn(*args, **kwargs)
  File "/home/fsp/.virtualenvs/nn/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 2369, in forward
    return compiled_fn(full_args)
  File "/home/fsp/.virtualenvs/nn/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 961, in new_fn
    fw_outs = call_func_with_args(compiled_fw, args, disable_amp=disable_amp)
  File "/home/fsp/.virtualenvs/nn/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py", line 917, in call_func_with_args
    out = normalize_as_list(f(args))
  File "/tmp/torchinductor_fsp/nx/cnxybmxjhsio5umlauprx5kj2eq6dnciwkzjaby334pqoiy6wiqn.py", line 81, in call
    assert_size_stride(buf3, (1, 16, 128, 144), (294912, 18432, 144, 1))
AssertionError: expected size 16==16, stride 1==18432 at dim=1
```

Related file in backtrace:

The expected stride of `buf3` is `(294912, 18432, 144, 1)`, but the actual stride `(294912, 1, 2304, 16)`, so the assertion fails. 

```python

from ctypes import c_void_p, c_long
import torch
import random
from torch import empty_strided, as_strided, device
from torch._inductor.codecache import AsyncCompile

aten = torch.ops.aten
assert_size_stride = torch._C._dynamo.guards.assert_size_stride
async_compile = AsyncCompile()


kernel_cpp_0 = async_compile.cpp('''
#include "/tmp/torchinductor_fsp/77/c7773nj5pwikpmm2pwa62rcudlf7p3if7eyqb5k4sjsvewwje4le.h"
extern "C" void kernel(const float* __restrict__ in_ptr0,
                       float* __restrict__ out_ptr0)
{
    #pragma omp parallel num_threads(6)
    {
        #pragma omp for 
        for(long i0=0; i0<16; i0+=1)
        {
            #pragma GCC ivdep
            for(long i1=0; i1<4608; i1+=1)
            {
                {
                    {
                        auto tmp0 = in_ptr0[i1 + (4608*i0)];
                        out_ptr0[i0 + (16*i1)] = tmp0;
                    }
                }
            }
        }
    }
}
''')


kernel_cpp_1 = async_compile.cpp('''
#include "/tmp/torchinductor_fsp/77/c7773nj5pwikpmm2pwa62rcudlf7p3if7eyqb5k4sjsvewwje4le.h"
extern "C" void kernel(float* __restrict__ in_out_ptr0)
{
    #pragma omp parallel num_threads(6)
    {
        #pragma omp for 
        for(long i0=0; i0<36864; i0+=1)
        {
            auto tmp0 = at::vec::Vectorized<float>::loadu(in_out_ptr0 + 8*i0);
            auto tmp1 = at::vec::clamp_min(tmp0, decltype(tmp0)(0));
            tmp1.store(in_out_ptr0 + 8*i0);
        }
        #pragma omp for simd simdlen(4) 
        for(long i0=294912; i0<294912; i0+=1)
        {
            auto tmp0 = in_out_ptr0[i0];
            auto tmp1 = tmp0 * (tmp0>0);
            in_out_ptr0[i0] = tmp1;
        }
    }
}
''')


async_compile.wait(globals())
del async_compile

def call(args):
    arg0_1, arg1_1, arg2_1 = args
    args.clear()
    buf0 = empty_strided((1, 16, 64, 72), (73728, 1, 1152, 16), device='cpu', dtype=torch.float32)
    kernel_cpp_0(c_void_p(arg2_1.data_ptr()), c_void_p(buf0.data_ptr()))
    del arg2_1
    buf1 = torch.ops.mkldnn._convolution_pointwise(buf0, arg0_1, arg1_1, (1, 1), (1, 1), (1, 1), 1, 'none', [], '')
    assert_size_stride(buf1, (1, 64, 64, 72), (294912, 1, 4608, 64))
    del arg0_1
    del arg1_1
    del buf0
    buf2 = torch.ops.aten.pixel_shuffle.default(buf1, 2)
    del buf1
    buf3 = buf2
    assert_size_stride(buf3, (1, 16, 128, 144), (294912, 18432, 144, 1))
    del buf2
    buf4 = buf3; del buf3  # reuse
    kernel_cpp_1(c_void_p(buf4.data_ptr()))
    return (buf4, )


if __name__ == "__main__":
    from torch._dynamo.testing import rand_strided
    from torch._inductor.utils import print_performance
    arg0_1 = rand_strided((64, 16, 3, 3), (1, 0, 0, 0), device='cpu', dtype=torch.float32)
    arg1_1 = rand_strided((64, ), (1, ), device='cpu', dtype=torch.float32)
    arg2_1 = rand_strided((1, 16, 64, 72), (73728, 4608, 72, 1), device='cpu', dtype=torch.float32)
    print_performance(lambda: call([arg0_1, arg1_1, arg2_1]))
```

### Versions

Collecting environment information...
PyTorch version: 2.0.0.dev20221231+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Arch Linux (x86_64)
GCC version: (GCC) 12.2.0
Clang version: 14.0.6
CMake version: version 3.25.1
Libc version: glibc-2.36

Python version: 3.10.9 (main, Dec 19 2022, 17:35:49) [GCC 12.2.0] (64-bit runtime)
Python platform: Linux-6.1.1-arch1-1-x86_64-with-glibc2.36
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.24.1
[pip3] torch==2.0.0.dev20221231+cpu
[pip3] torchaudio==2.0.0.dev20221230+cpu
[pip3] torchvision==0.15.0.dev20221230+cpu
[conda] No relevant packages

cc @ezyang @gchanan @zou3519 @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

**Comment by anijain2305** [comment]
cc @jgong5

**Comment by ezyang** [comment]
hooray for asserts lol

**Comment by Xia-Weiwen** [comment]
Fixed in https://github.com/pytorch/pytorch/pull/91605

**Comment by soumith** [comment]
fixed via #91605

**Comment by yuanzhi-zhu** [comment]
Got a similar error with torch 2.2.0, and the error disappeared after upgrading torch to 2.3.0
